SU-HSDO/suhumsci
# STN-691: Prevent Bullet List Numbers from Overlapping when there is Multiple Digits

# [STN-691](https://sparkbox.atlassian.net/browse/STN-691) READY FOR REVIEW

## Summary
The work in this PR updates the minimum width of list markers to prevent the digits in ordered lists from overlapping the content in the list.

This work also updates the alignment of the list markers to align to the right.

## Need Review By (Date)
3/3

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. To test list marker style updates, sync your local environment with staging to pull in the [QA List Marker Styles](https://sparkbox-sandbox-stage.stanford.edu/qa-list-marker-styles) page. Or you can create a new flexible page, add a new text area, and copy/paste the following code block:
```
<h2>Default List Styles</h2>

<h3>Unordered List</h3>

<ul>
	<li>partridge in a pear tree</li>
	<li>turtle doves</li>
	<li>french hens
	<ul>
		<li>calling birds</li>
		<li>gold rings</li>
		<li>geese a-laying
		<ul style="list-style-type: circle;">
			<li>swans a-swimming</li>
			<li>maid a-milking</li>
			<li>ladies dancing</li>
		</ul>
		</li>
		<li>lords a-leaping</li>
		<li>pipers piping</li>
	</ul>
	</li>
	<li>drummers drumming</li>
	<li>starlings</li>
</ul>

<h3>Ordered List</h3>

<ol start="1">
	<li>partridge in a pear tree</li>
	<li>turtle doves</li>
	<li>french hens</li>
	<li>turkeys</li>
	<li>chickens</li>
	<li>ducks</li>
	<li>turtles</li>
	<li>cats</li>
	<li>dogs</li>
	<li>rabbits</li>
	<li>zebras</li>
	<li>horses</li>
	<li>elephants</li>
	<li>cows</li>
	<li>sheep</li>
	<li>butterflies</li>
	<li>bumble bees</li>
	<li>goats</li>
	<li>monkeys</li>
	<li>lions</li>
	<li>tigers</li>
	<li>skunk
	<ol>
		<li>calling birds</li>
		<li>gold rings</li>
		<li>geese a-laying
		<ol>
			<li>swans a-swimming</li>
			<li>maid a-milking</li>
			<li>ladies dancing</li>
			<li>walking</li>
			<li>running</li>
			<li>jumping</li>
			<li>yoga-ing</li>
			<li>crossfitting</li>
			<li>weight lifting</li>
			<li>hiking</li>
			<li>cross training</li>
			<li>surfing</li>
			<li>crawling</li>
			<li>eating</li>
			<li>sleeping</li>
			<li>drinking</li>
			<li>jumping</li>
			<li>singing</li>
			<li>painting</li>
			<li>dancing</li>
		</ol>
		</li>
		<li>lords a-leaping</li>
		<li>pipers piping</li>
	</ol>
	</li>
	<li>drummers drumming</li>
	<li>starlings</li>
</ol>
```

And this code block of mixed lists in another text area:
```
<h2>Mixed Lists</h2>

<h3>ol ol ul</h3>

<ol>
	<li>You better watch out</li>
	<li>You better not cry</li>
	<li>Better not pout
	<ol>
		<li>I'm telling you why</li>
		<li>Santa Claus is coming to town</li>
		<li>He's making a list</li>
		<li>And checking it twice;
		<ul>
			<li>Gonna find out</li>
			<li>Who's naughty and nice</li>
			<li>Santa Claus is coming to town</li>
		</ul>
		</li>
	</ol>
	</li>
</ol>

<h3>ol ul ol</h3>

<ol>
	<li>Frosty the snowman</li>
	<li>Was a jolly happy soul</li>
	<li>With a corncob pipe
	<ul>
		<li>And a button nose and</li>
		<li>Two eyes made out of coal</li>
		<li>Frosty the snowman</li>
		<li>Is a fairy tale they say
		<ol>
			<li>He was made of snow</li>
			<li>But the children know</li>
			<li>How he came to life one day</li>
		</ol>
		</li>
	</ul>
	</li>
</ol>

<h3>ol ul ul</h3>

<ol>
	<li>There must have been some magic</li>
	<li>In that old silk hat they found</li>
	<li>For when they placed it on his head
	<ul>
		<li>He began to dance around</li>
		<li>Frosty the snowman</li>
		<li>Was alive as he could be
		<ul>
			<li>And the children say</li>
			<li>He could laugh and play</li>
			<li>Just the same as you and me</li>
		</ul>
		</li>
	</ul>
	</li>
</ol>

<h3>ul ol ol</h3>

<ul>
	<li>Frosty the snowman</li>
	<li>Knew the sun was hot that day</li>
	<li>So he said "Let's run
	<ol>
		<li>And we'll have some fun</li>
		<li>Now before I melt away"</li>
		<li>Down to the village</li>
		<li>With a broomstick in his hand
		<ol>
			<li>Running here and there</li>
			<li>All around the square</li>
			<li>Saying "Catch me if you can"</li>
		</ol>
		</li>
	</ol>
	</li>
</ul>

<h3>ul ul ol</h3>

<ul>
	<li>He led them down the streets of town</li>
	<li>Right to the traffic cop</li>
	<li>And he paused a moment when
	<ul>
		<li>He heard him holler "Stop"</li>
		<li>Frosty the snowman
		<ol>
			<li>Had to hurry on his way</li>
			<li>But he waved goodbye saying,</li>
		</ol>
		</li>
	</ul>
	</li>
</ul>

<h3>ul ol ul</h3>

<ul>
	<li>"Don't you cry</li>
	<li>I'll be back again some day"</li>
	<li>Thumpity, thump, thump
	<ol>
		<li>Thumpity, thump, thump</li>
		<li>Look at Frosty go
		<ul>
			<li>Thumpity, thump, thump</li>
			<li>Thumpity, thump, thump</li>
			<li>Over the hills of snow</li>
		</ul>
		</li>
	</ol>
	</li>
</ul>
```
3. - [x] Confirm ordered list items support up to two characters / up to 99 list items without text overflow. Note that spacing has been set to accommodate the roman numeral 18 (xviii) which appears to be the widest potential number.
4.  - [x] Confirm the utility class of `hb-list-reset` continues to reset list styles to browser defaults. And that bulleted list properties can be set in the WYSIWYG when applied.
5. - [x] Confirm unordered list styles are unaffected
6. - [x] Confirm ordered lists still work when next to an image that is floated to the left or right.
